lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnJobSubmissionHandler.java

```diff
@@ -69,7 +72,8 @@
       final ClasspathProvider classpath,
       final ConfigurationSerializer configurationSerializer,
       final JobUploader uploader,
-      @Parameter(JVMHeapSlack.class) final double jvmSlack) throws IOException {
+      @Parameter(JVMHeapSlack.class) final double jvmSlack,
+      @Parameter(JobQueue.class) String defaultQueueName) throws IOException {
```

Comment:
SO currently when we do not specify any queue to yarn it is able to find
the default queue right on MSR cluster. Can't we maintain the same logic
here?
On Jul 7, 2015 08:08, "Markus Weimer" <notifications@github.com> wrote:

> In
> lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnJobSubmissionHandler.java
> <https://github.com/apache/incubator-reef/pull/279#discussion_r34050474>:
>
> > @@ -69,7 +72,8 @@
> >        final ClasspathProvider classpath,
> >        final ConfigurationSerializer configurationSerializer,
> >        final JobUploader uploader,
> > -      @Parameter(JVMHeapSlack.class) final double jvmSlack) throws IOException {
> > +      @Parameter(JVMHeapSlack.class) final double jvmSlack,
> > +      @Parameter(JobQueue.class) String defaultQueueName) throws IOException {
>
> It is given by the user, but has a default value, 'default'. That is the
> *only* queue the current code submits to. I don't know whether YARN has
> an API to figure out the default queue. If we want to use it, I think it
> would be a good idea to do that in a separate JIRA, as it would actually
> change current behavior without application changes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/279/files#r34050474>.
>